vispy/app/backends/_pyqt6.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) Vispy Development Team. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+
+"""PyQt6 proxy backend for the qt backend.
+Based on PySIde6 backend.
+"""
+
+import sys
+from .. import backends
+from ...util import logger
+from ... import config
+
+USE_EGL = config['gl_backend'].lower().startswith('es')
+
+try:
+    # Make sure no conflicting libraries have been imported.
+    for lib in ['PySide', 'PySide2', 'PySide6', 'PyQt4', 'PyQt5']:
+        lib += '.QtCore'
+        if lib in sys.modules:
+            raise RuntimeError("Refusing to import PyQt6 because %s is "
+                               "already imported." % lib)
+    # Try importing (QtOpenGL first to fail without import QtCore)
+    if not USE_EGL:
+        from PyQt6 import QtOpenGL  # noqa
+    from PyQt6 import QtGui, QtCore  # noqa
+except Exception as exp:
+    # Fail: this backend cannot be used
+    available, testable, why_not, which = False, False, str(exp), None
+else:
+    # Success
+    available, testable, why_not = True, True, None
+    # What is this for? PyQt5 sets this to true, PySide sets this to false
+    has_uic = False
+
+    which = ('PyQt6', QtCore.PYQT_VERSION_STR, QtCore.PYQT_VERSION_STR)
+    # Remove _qt module to force an import even if it was already imported
+    sys.modules.pop(__name__.replace('_pyqt6', '_qt'), None)
+    # Import _qt. Keep a ref to the module object!
+    if backends.qt_lib is None:
+        backends.qt_lib = 'pyqt6'  # Signal to _qt what it should import
+        from . import _qt  # noqa
+        from ._qt import *  # noqa
```

Comment:
If @mars0001 is up for it ... I think the way to go (to clean up these  `sys.modules` checks) would be to add a `def _check_imported(current_lib)` in `_qt.py` that we call right here.

Reply:
@almarklein, Sure, once I get home from work I will do it!

Reply:
We could also in the long run switch to `qtpy` which @sofroniewn and maybe @tlambert03 have suggested in the past: https://pypi.org/project/QtPy/

Reply:
Wait, such a function already exists, and is already called in `_qt.py` itself. It looks to me like the import-checks in all qt backends (except the one one in `_qt.py` can be removed ...

Reply:
> We could also in the long run switch to qtpy

which also doesn't support pyqt6 yet 😞 ...  there's a seemingly now stale thread there from Dec 2020 to add support for pyside6 and pyqt6. we have an independent compatibility layer ([superqt.qtcompat](https://github.com/napari/superqt/tree/main/superqt/qtcompat)) that has qt6 support